### PR TITLE
adding 3 keys {ravindra, kb, master} to keys.coffee

### PR DESCRIPTION
**Pull request by subham103**
@hritvi  please review.

### CODE REVIEW

**Comment by IronicallySerious** [comment]
Can you change the command usage format as listed here https://github.com/sdslabs/bot/blob/9ca53328054128b389918b9789b8e52b70369c6d/scripts/keys.coffee#L5 ?

File: scripts/keys.coffee [hunk 1]
```diff
@@ -22,81 +22,213 @@
...
 		name = msg.message.user.name 
 		user = robot.brain.userForName name
-		k = key()
-		if typeof user is 'object'
-			msg.send "Okay #{name} has keys"
-			k = user.name
-		robot.brain.set("key",k)	
-
+		if keyname is 'ravindra' or 'rv'
+			k = ravindra_key()
```
Comment (IronicallySerious) [inline scripts/keys.coffee]: Prefer a more descriptive variable name

File: scripts/keys.coffee [hunk 2]
```diff
@@ -22,81 +22,213 @@
...
+	ravindra_key = ()->
+		Key = robot.brain.get("ravindra_key") or ""
+		robot.brain.set("ravindra_key" ,Key)
+		Key
+
+	master_key = ()->
+		Key = robot.brain.get("master_key") or ""
+		robot.brain.set("master_key" ,Key)
+		Key
+
```
Comment (IronicallySerious) [inline scripts/keys.coffee]: Remove extra newline from here. Only one is required :)

File: scripts/keys.coffee [hunk 3]
```diff
@@ -22,81 +22,213 @@
...
+			robot.brain.set("kb_key",k)	
+		else if keyname is 'master'
+			k = master_key()
+			if k is name
+				k = ""
+			if typeof user is 'object'
+				if k is ""
+					msg.send "Okay #{name} doesn't have master keys. Who got the master keys then?"
+				else
+					msg.send "Yes , i know buddy, its because #{k} has got the master keys"
```
Comment (IronicallySerious) [inline scripts/keys.coffee]: Capitalise the `i` in `i know buddy`

File: scripts/keys.coffee [hunk 4]
```diff
@@ -22,81 +22,213 @@
...
-		k = key()
-		if typeof user is 'object'
-			msg.send "Okay #{name} has keys"
-			k = user.name
-		robot.brain.set("key",k)	
-
+		if keyname is 'ravindra' or 'rv'
+			k = ravindra_key()
+			if typeof user is 'object'
+				msg.send "Okay #{name} has #{keyname} keys"
```
Comment (IronicallySerious) [inline scripts/keys.coffee]: See how you can optimize these if statements. This code segment...
```coffee
msg.send "Okay #{name} has #{keyname} keys"
  robot.brain.set("key",k)						
  k = user.name
  robot.brain.set("ravindra_key",k)
```
...is being repeated 3 times with very little changes

File: scripts/keys.coffee [hunk 5]
```diff
@@ -22,81 +22,213 @@
...
-		if typeof user is 'object'
-			if k is ""
-				msg.send "Okay #{name} doesn't have keys. Who got the keys then?"
-			else
-				msg.send "Yes , i know buddy, its because #{k} has got the keys"	
-		robot.brain.set("key",k)	
-
+		
+		if keyname is 'ravindra' || keyname is 'rv'
+			k = ravindra_key()
```
Comment (IronicallySerious) [inline scripts/keys.coffee]: Same goes here. Repetition of code means that something can be done even better.

File: scripts/keys.coffee [hunk 6]
```diff
@@ -22,81 +22,213 @@
...
+#this section belongs to "$name has keys"
+	robot.respond /(.+) (has|have) (.+) (the key|key|keys|a key)/i , (msg)->
 		othername = msg.match[1]
+		keyname = msg.match[3]
 		name = msg.message.user.name
-		k = key()
-		unless othername in ["who", "who all","Who", "Who all" , "i" , "I" , "i don't" , "i dont" , "i do not" , "I don't" , "I dont" , "I do not"]
+		
+		if keyname is 'ravindra' || keyname is 'rv'
+			k = ravindra_key()
```
Comment (IronicallySerious) [inline scripts/keys.coffee]: This also looks like the code is being repeated. If you find a solution to remove repetition in the above lines then a similar strategy will work here too.

File: scripts/keys.coffee [hunk 7]
```diff
@@ -22,81 +22,213 @@
...
+				else
+					k = users.name
+					msg.send "Okay, so now the master keys are with #{users.name}"	
+			robot.brain.set("master_key",k)		
+		
+#this section is to print the details about the key holder.		
+	robot.respond /(who|who all) (has|have|has the|have the|has a|have a) (.+) (key|keys)/i , (msg)->
+		othername = msg.match[3]
+		if othername is 'ravindra' || othername is 'rv'
+			k1 = ravindra_key()
```
Comment (IronicallySerious) [inline scripts/keys.coffee]: Prefer better names than `k1`, `k2` and `k3` :P 
Also, the code block below can optimised too

File: scripts/keys.coffee (hunk 1 above)
Comment (subham103) [reply scripts/keys.coffee]: ok @IronicallySerious .. I replaced 'k' with 'key_holder'.

File: scripts/keys.coffee (hunk 3 above)
Comment (subham103) [reply scripts/keys.coffee]: replaced i with I :)

File: scripts/keys.coffee (hunk 7 above)
Comment (subham103) [reply scripts/keys.coffee]: @IronicallySerious , @hritvi !! please review the changes..

**Comment by hritvi** [comment]
@subham103 remove the package-lock.json. Rest looks good to me!!

**Pull request merged**